logstash-core/lib/logstash/pipeline.rb

```diff
@@ -18,18 +18,21 @@
 require "logstash/instrument/namespaced_null_metric"
 require "logstash/instrument/collector"
 require "logstash/instrument/wrapped_write_client"
+require "logstash/util/dead_letter_queue_manager"
 require "logstash/output_delegator"
 require "logstash/filter_delegator"
 require "logstash/queue_factory"
 require "logstash/compiler"
 require "logstash/execution_context"
 
+java_import org.logstash.common.io.DeadLetterQueueWriter
+
 module LogStash; class BasePipeline
   include LogStash::Util::Loggable
 
   attr_reader :config_str, :config_hash, :inputs, :filters, :outputs, :pipeline_id, :lir
   
-  def initialize(config_str, settings = SETTINGS)
+  def initialize(config_str, settings = SETTINGS, for_validation = false)
```

Comment:
I think we discussed this in the meeting, but I will note the same here.

If we have a concept of "validating a pipeline" we should separate that concern somewhere else. A boolean that significantly alters the behavior of the pipeline feels quite odd, and I think it needs separation into a different class.